src/index.js

```diff
@@ -111,17 +111,30 @@ class EventsCoreAPI {
    * =========================================================================
    */
 
+  /**
+   * @typedef {object} ProviderOptions
+   * @property {string} [providerMetadataId] Fetch by providerMetadataId for the consumer org
+   * @property {string} [instanceId] For Self registered providers, instanceId is a must while fetching by providerMetadataId
+   * @property {Array} [providerMetadataIds] Fetch all providers ( and all instances ) for the list of provider metadata ids
+   */
   /**
    * Fetch all the providers
    *
    * @param {string} consumerOrgId Consumer Org Id from the console
+   * @param {boolean} fetchEventMetadata Option to fetch event metadata for each of the the providers in the list
+   * @param {ProviderOptions} providerOptions Provider filtering options based on either (providerMetadataId and instanceId) or list of providerMetadataIds
    * @returns {Promise<object>} Returns list of providers for the org
    */
-  getAllProviders (consumerOrgId) {
+  getAllProviders (consumerOrgId, fetchEventMetadata = false, providerOptions = {}) {
```

Comment:
instead of two params `fetchEventMetadata` and `providerOptions` which are both configuration type params can we merge them into a single `options` params? some thing like `options.fetchEventMetadata` and `options.providerOptions`

Reply:
i did think about the same. but providerOptions is more of filtering criteria, where as fetchEventMetadata is about adding event metadata to the response returned. do you still think it should be a single options param?

Reply:
I also kind of dislike booleans as parameters, I would advise for the same, maybe having
`options.fetchEventMetadata` and `options.filterBy` so that is a bit more clear maybe

Reply:
so an object like 
{
    "fetchEventMetadata": true/false,
    "filterBy": {
        "providerMetadataId":"",
        "instanceId":"",
        "providerMetadataIds":""
    }
}

Reply:
yes, how do you like it?

Reply:
looks good

Reply:
done